R/tune_bayes.R

```diff
@@ -211,7 +211,8 @@ tune_bayes.model_spec <- function(object,
     ))
   }
 
-  control <- parsnip::condense_control(control, control_bayes())
+  # set `seed` so that calling `control_bayes()` doesn't alter RNG state (#721)
+  control <- parsnip::condense_control(control, control_bayes(seed = 1))
```

Comment:
Does not change behavior of the check, but note that:

``` r
library(tune)

set.seed(1)

head(.Random.seed)
#> [1]      10403        624 -169270483 -442010614 -603558397 -222347416

ctrl <- control_bayes()

# state of RNG changes:
head(.Random.seed)
#> [1]       10403           2  1654269195 -1877109783  -961256264  1403523942

# but, if we set the seed...
ctrl_seeded <- control_bayes(seed = 1)

head(.Random.seed)
#> [1]       10403           2  1654269195 -1877109783  -961256264  1403523942
```

<sup>Created on 2023-10-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
This is very counter-intuitive usage of the `seed` argument: I would expect specifying a `seed` argument to result in changes to `.Random.seed` and every call to `control_bayes()` with the same value for the `seed` argument to have the same result aka state of `.Random.seed`. Instead, setting the `seed` argument results in kind of the opposite: `.Random.seed` is _unchanged_ and _independent_ of what value I use for the `seed` argument:

``` r
library(tune)

set.seed(1)

head(.Random.seed)
#> [1]      10403        624 -169270483 -442010614 -603558397 -222347416

ctrl <- control_bayes()

# state of RNG changes:
head(.Random.seed)
#> [1]       10403           2  1654269195 -1877109783  -961256264  1403523942

# but, if we set the seed...
ctrl_seeded <- control_bayes(seed = 1)

head(.Random.seed)
#> [1]       10403           2  1654269195 -1877109783  -961256264  1403523942

# and the value for `seed` does not matter in this regard
ctrl_seeded <- control_bayes(seed = 2)

head(.Random.seed)
#> [1]       10403           2  1654269195 -1877109783  -961256264  1403523942
```

<sup>Created on 2023-10-03 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

I get why you are doing that here but I think it would be worthwhile to investigate if we want to change the behavior of `control_bayes()` in that regard.